docs/specifications/tests/DNSSEC-TP/dnssec11.md

```diff
@@ -1,50 +1,244 @@
 
-If there is a DS record that matches a DNSKEY record that correctly signs the
-apex DNSKEY RRset, the test case passes.
+## Scope
 
-### Special procedural requirements
+It is assumed that *Child Zone* is tested and reported by other test cases:
 
-None.
+* This test case will just ignore non-responsive name servers or name servers
+  not giving a correct DNS response for an authoritative name server, covered
+  by [Basic04].
+* This test case will ignore any irregularities in fetching the DS record from
+  parent zoen, covered by [DNSSEC02].
+* This test case will ignore if the DNSKEY and SOA RRsets are signed, covered
+  by [DNSSEC08] and [DNSSEC09].
+
+## Inputs
+
+* "Child Zone" - The domain name to be tested.
+* "Test Type" - The test type with value "[undelegated]" or "normal".
+* "Undelegated DS" - The DS record or records submitted
+  (only if *Test Type* is [undelegated]).
+
+
+## Summary
+
+* 
+
+Message Tag outputted         | Level   | Arguments  | Description of when message tag is outputted
+:-----------------------------|:--------|:-----------|:--------------------------------------------
+DS11_INCONSISTENT_DS          | WARNING |            | Parent name servers are inconsistent on the existence of DS.
+DS11_INCONSISTENT_SIGNED_ZONE | ERROR   |            | Name servers for the child zone is inconsitent on wherer the zone is signed or not.
+DS11_INDETERMINED_DS          | ERROR   |            | It cannot be determined if the parent zone as DS for the child zone or not.
+DS11_INDETERMINED_SIGNED_ZONE | ERROR   |            | It cannot be determined if the child zone is signed or not.
+DS11_PARENT_WITHOUT_DS        | NOTICE  | ns_ip_list | List of parent name servers without DS for the child zone.
+DS11_PARENT_WITH_DS           | NOTICE  | ns_ip_list | List of parent name servers with DS for the child zone.
+DS11_NS_WITH_SIGNED_ZONE      | NOTICE  | ns_ip_list | List of child name servers with signed child zone.
+DS11_NS_WITH_UNSIGNED_ZONE    | WARNING | ns_ip_list | List of child name servers with unsigned child zone.
+DS11_DS_BUT_UNSIGNED_ZONE     | ERROR   |            | The child zone is unsigned, but the parent zone has DS record.
+
+The value in the Level column is the default severity level of the message. The
+severity level can be changed in the [Zonemaster-Engine profile]. Also see the
+[Severity Level Definitions] document.
+
+The argument names in the Arguments column lists the arguments used in the
+message. The argument names are defined in the [argument list].
+
+
+## Test procedure
+
+1.  Create the following empty sets:
+    1. Parent name server IP address ("Indetermined DS").
+    2. Parent name server IP address ("No DS Record").
+    3. Parent name server IP address ("Has DS Record").
+    4. Child name server IP address ("Indeterminded DNSKEY")
+    5. Child name server IP address ("No DNSKEY Record").
+    6. Child name server IP address ("Has DNSKEY Record").
+
+2.  If *Test Type* is "normal", then:
```

Comment:
Could you swap 2. and 3. ?

Reply:
> Both orders work.

Yes totally.
It's just that I think it helps to get rid of "small" conditions first before going into the heavy work.